2.4/Dockerfile.c9s

```
@@ -0,0 +1,66 @@
+FROM quay.io/sclorg/s2i-core-c9s:c9s
+
+# Apache HTTP Server image.
+#
+# Volumes:
+#  * /var/www - Datastore for httpd
+#  * /var/log/httpd24 - Storage for logs when $HTTPD_LOG_TO_VOLUME is set
+# Environment:
+#  * $HTTPD_LOG_TO_VOLUME (optional) - When set, httpd will log into /var/log/httpd24
+
+ENV HTTPD_VERSION=2.4
+
+ENV SUMMARY="Platform for running Apache httpd $HTTPD_VERSION or building httpd-based application" \
+    DESCRIPTION="Apache httpd $HTTPD_VERSION available as container, is a powerful, efficient, \
+and extensible web server. Apache supports a variety of features, many implemented as compiled modules \
+which extend the core functionality. \
+These can range from server-side programming language support to authentication schemes. \
+Virtual hosting allows one Apache installation to serve many different Web sites."
+
+LABEL summary="$SUMMARY" \
+      description="$DESCRIPTION" \
+      io.k8s.description="$DESCRIPTION" \
+      io.k8s.display-name="Apache httpd $HTTPD_VERSION" \
+      io.openshift.expose-services="8080:http,8443:https" \
+      io.openshift.tags="builder,httpd,httpd-24" \
+      name="sclorg/httpd-24-c9s" \
+      version="1" \
+      com.redhat.license_terms="https://www.redhat.com/en/about/red-hat-end-user-license-agreements#rhel" \
+      com.redhat.component="httpd-24-container" \
+      usage="s2i build https://github.com/sclorg/httpd-container.git --context-dir=examples/sample-test-app/ quay.io/sclorg/httpd-24-c9s sample-server" \
+      maintainer="SoftwareCollections.org <sclorg@redhat.com>"
+
+EXPOSE 8080
+EXPOSE 8443
+
+RUN INSTALL_PKGS="gettext hostname nss_wrapper bind-utils httpd mod_ssl mod_ldap mod_session mod_security mod_auth_mellon sscg" && \
```

```suggestion
yum -y module enable httpd:$HTTPD_VERSION && \
RUN INSTALL_PKGS="gettext hostname nss_wrapper bind-utils httpd mod_ssl mod_ldap mod_session mod_security mod_auth_mellon sscg" && \
```
centos7 Dockerfile also enables module httpd with correct version of httpd, isn't this or some equivalent needed here?

---

More specifically, I understand, that the current version of httpd in c9s is 2.4 and it is just the one, which is needed here, but couldn't this be changed later so that a bug is introduced?

---

I think, that this question will appear in more c9s support PRs, as I quickly checked the MariaDB container.

---

As far as I know, modules are not supported in CentOS Stream 9.
@pkubatrh Am I right?

---

Then the solution can be to  install `httpd-$HTTPD_VERSION*` instead of `httpd`.
But I also agree if you say, that this isn't a problem at all and versioning is unnecessary here.
WDYT?

---

Because of there are no modules, this is not problem. Soon we will switch back to the modules.
versioning is not necessary at all. we have only one version.